packages/web3-eth/src/utils/detect_transaction_type.ts

```diff
@@ -17,26 +17,102 @@ along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
 
 import { format, toHex } from 'web3-utils';
 import { TransactionTypeParser, Web3Context } from 'web3-core';
-import { EthExecutionAPI, Transaction, ETH_DATA_FORMAT } from 'web3-types';
-import { isNullish } from 'web3-validator';
+import { EthExecutionAPI, HardforksOrdered, Transaction, ETH_DATA_FORMAT } from 'web3-types';
+import { isNullish, validator } from 'web3-validator';
 import { InternalTransaction } from '../types';
 
+// undefined is treated as null for JSON schema validator
+const type0x0TransactionSchema = {
+	type: 'object',
+	properties: {
+		accessList: {
+			type: 'null',
+		},
+		maxFeePerGas: {
+			type: 'null',
+		},
+		maxPriorityFeePerGas: {
+			type: 'null',
+		},
+	},
+};
+const type0x1TransactionSchema = {
+	type: 'object',
+	properties: {
+		maxFeePerGas: {
+			type: 'null',
+		},
+		maxPriorityFeePerGas: {
+			type: 'null',
+		},
+	},
+};
+const type0x2TransactionSchema = {
+	type: 'object',
+	properties: {
+		gasPrice: {
+			type: 'null',
+		},
+	},
+};
+
```

Comment:
I don't believe these should go in [schemas.ts](https://github.com/web3/web3.js/blob/4.x/packages/web3-eth/src/schemas.ts) since they aren't complete schemas of the transaction object

Reply:
@jdevcs 
I'll rename them to: `transactionType0x0Schema`, but I don't think they should be moved to `schemas.ts` because then they'll be exported from `web3-eth` package and these type shouldn't be any where else as they are an incomplete representation of what a typed transaction is (i.e. they don't contain the other fields of a transaction)

I experimented with doing something like:

```typescript
export const transactionType0x0Schema = {
	type: 'object',
	properties: {
		...transactionSchema.properties,
		accessList: {
			type: 'null',
		},
		maxFeePerGas: {
			type: 'null',
		},
		maxPriorityFeePerGas: {
			type: 'null',
		},
	}
};
```

but now the function `validateTxTypeAndHandleErrors` in `defaultTransactionTypeParser` is validating the entire transaction object which is out of scope of `defaultTransactionTypeParser`. This function is only concerned about returning the detected tx type, or throwing an error for an invalid tx object regarding types and their appropriate properties, not the rest of the tx object

I think it's okay to keep these type specified here so that they're are exported from the package and because they are only used for `defaultTransactionTypeParser`

Reply:
> because then they'll be exported from web3-eth package

that make sense to not export it, with current code that you added.